{
 "cells": [
  {
   "cell_type": "code",
   "execution_count": 30,
   "id": "906518ec",
   "metadata": {},
   "outputs": [
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "C:\\Users\\ANDREC~1\\AppData\\Local\\Temp/ipykernel_23240/2927581491.py:56: SettingWithCopyWarning: \n",
      "A value is trying to be set on a copy of a slice from a DataFrame.\n",
      "Try using .loc[row_indexer,col_indexer] = value instead\n",
      "\n",
      "See the caveats in the documentation: https://pandas.pydata.org/pandas-docs/stable/user_guide/indexing.html#returning-a-view-versus-a-copy\n",
      "  df1['#CHROM'] = df1['chrom']\n",
      "C:\\Users\\ANDREC~1\\AppData\\Local\\Temp/ipykernel_23240/2927581491.py:57: SettingWithCopyWarning: \n",
      "A value is trying to be set on a copy of a slice from a DataFrame.\n",
      "Try using .loc[row_indexer,col_indexer] = value instead\n",
      "\n",
      "See the caveats in the documentation: https://pandas.pydata.org/pandas-docs/stable/user_guide/indexing.html#returning-a-view-versus-a-copy\n",
      "  df1['POS'] = df1['position']\n",
      "C:\\Users\\ANDREC~1\\AppData\\Local\\Temp/ipykernel_23240/2927581491.py:58: SettingWithCopyWarning: \n",
      "A value is trying to be set on a copy of a slice from a DataFrame.\n",
      "Try using .loc[row_indexer,col_indexer] = value instead\n",
      "\n",
      "See the caveats in the documentation: https://pandas.pydata.org/pandas-docs/stable/user_guide/indexing.html#returning-a-view-versus-a-copy\n",
      "  df1['REF'] = df1['ref']\n",
      "C:\\Users\\ANDREC~1\\AppData\\Local\\Temp/ipykernel_23240/2927581491.py:59: SettingWithCopyWarning: \n",
      "A value is trying to be set on a copy of a slice from a DataFrame.\n",
      "Try using .loc[row_indexer,col_indexer] = value instead\n",
      "\n",
      "See the caveats in the documentation: https://pandas.pydata.org/pandas-docs/stable/user_guide/indexing.html#returning-a-view-versus-a-copy\n",
      "  df1['ALT'] = df1['var']\n",
      "C:\\Users\\ANDREC~1\\AppData\\Local\\Temp/ipykernel_23240/2927581491.py:60: SettingWithCopyWarning: \n",
      "A value is trying to be set on a copy of a slice from a DataFrame.\n",
      "Try using .loc[row_indexer,col_indexer] = value instead\n",
      "\n",
      "See the caveats in the documentation: https://pandas.pydata.org/pandas-docs/stable/user_guide/indexing.html#returning-a-view-versus-a-copy\n",
      "  df1['somatic_p_value'] = df1['somatic_p_value'].astype(float)\n"
     ]
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "done\n"
     ]
    }
   ],
   "source": [
    "# Full Python Script\n",
    "# Python 3.8.8\n",
    "# Pandas 1.2.4\n",
    "# Numpy 1.20.1\n",
    "# Carson Broeker, 10/19/2021\n",
    "#Test script for converting Mutect, SomaticSniper, and Varscan SNV vcf data to a list of genes\n",
    "# First, import necessary modules\n",
    "import math\n",
    "import pandas as pd\n",
    "import numpy as np\n",
    "import sys\n",
    "import csv\n",
    "from functools import reduce\n",
    "\n",
    "# Make it so really large files don't cause the system to error out\n",
    "maxInt = sys.maxsize\n",
    "\n",
    "while True:\n",
    "    # decrease the maxInt value by factor 10 \n",
    "    # as long as the OverflowError occurs.\n",
    "\n",
    "    try:\n",
    "        csv.field_size_limit(maxInt)\n",
    "        break\n",
    "    except OverflowError:\n",
    "        maxInt = int(maxInt/10)\n",
    "# Next, read in all of the VCF files using a for loop\n",
    "# This step also gets rid of superfluous columns for downstream analyses\n",
    "# Modify the List elements to suit your project needs\n",
    "# Make sure all of your vcf files are in the same directory/location as this python script\n",
    "#dfFVB_Var = pd.read_csv(\"FVB_NJVarscanSNVs.vcf.snp\", sep='\\t', usecols=[0,1,2,3,12,14], engine = 'c', dtype=object)\n",
    "#dfFVB_Var = dfFVB_Var[(dfFVB_Var['somatic_status'] == 'Somatic')]\n",
    "#dfFVB_Var['#CHROM'] = dfFVB_Var['chrom']\n",
    "#dfFVB_Var['POS'] = dfFVB_Var['position']\n",
    "#dfFVB_Var['REF'] = dfFVB_Var['ref']\n",
    "#dfFVB_Var['ALT'] = dfFVB_Var['var']\n",
    "\n",
    "#dfFVB_Som = pd.read_csv(\"FVB_NJSomaticSniperCalls.vcf\", sep='\\t', skiprows=17, usecols=[0,1,3,4], engine = 'c', dtype=object)\n",
    "\n",
    "df_Sanger_FVB_Germ = pd.read_csv(\"FVB_NJ.mgp.v5.snps.dbSNP142.vcf\", sep='\\t', skiprows=68, usecols=[0,1,3,4], engine='c', dtype=object)\n",
    "\n",
    "df_Haplotype = pd.read_csv(\"FVB_NJ_HaplotypeCaller.vcf\", sep='\\t', skiprows=93, usecols=[0,1,3,4], engine='c', dtype=object)\n",
    "\n",
    "Myc_List = ['TA39_927','TA41_1938','WT13_842','WT13_864','WT13_1356','WT13_1445','WT13_1576','WT21_222','WT21_1066']\n",
    "#Myc_List = ['WT13_1356']\n",
    "for base in Myc_List:\n",
    "    Varscan_base = pd.read_csv(base+\"VarscanSNVs.vcf.snp\", sep='\\t', usecols=[0,1,2,3,12,14], \\\n",
    "                             engine = 'c', dtype=object)\n",
    "# Must be loaded as engine='c' as the 'python' engine isn't natively equipped with the necessary modules\n",
    "# Next, make each dataframe something easier to read and obvious\n",
    "# If unsure how to modify data, look at the python pandas documentation\n",
    "    df1 = Varscan_base\n",
    "# Use masking to include only columns that have deletions or duplications, no inversions or translocations\n",
    "# Check CNV_Filtering python file for more advanced documentation\n",
    "    df1 = df1[(df1['somatic_status'] == 'Somatic')]\n",
    "    df1['#CHROM'] = df1['chrom']\n",
    "    df1['POS'] = df1['position']\n",
    "    df1['REF'] = df1['ref']\n",
    "    df1['ALT'] = df1['var']\n",
    "    df1['somatic_p_value'] = df1['somatic_p_value'].astype(float)\n",
    "    #df1FVB_Var = pd.merge(df1, dfFVB_Var, on=['#CHROM','POS','REF','ALT'])\n",
    "    #df1new = pd.concat([df1, df1FVB_Var]).drop_duplicates(subset=['#CHROM','POS','REF','ALT'], keep=False)\n",
    "    \n",
    "    Mutect_base = pd.read_csv(base+\"Mutect2Annotated.vcf\", sep='\\t', skiprows=106, usecols=[0,1,3,4,7], \\\n",
    "                             engine = 'c', dtype=object)\n",
    "    df2 = Mutect_base\n",
    "    #df2 = df2[df2['INFO'].str.contains('HIGH|MODERATE|LOW')]\n",
    "    #df2['READ_DEPTH'] = df2['INFO'].str.extract('DP=(\\d+);').astype(int)\n",
    "    \n",
    "    #SomaticSniper_base = pd.read_csv(base+\"_MycSomaticSniperCallsAnnotated.vcf\", sep='\\t', skiprows=22, usecols=[0,1,3,4,7], engine = 'c', dtype=object)\n",
    "    #df3 = SomaticSniper_base\n",
    "    #df3 = df3[df3['INFO'].str.contains('HIGH|MODERATE|LOW')]\n",
    "    #df3FVB_Som = pd.merge(df3, dfFVB_Som, on=['#CHROM','POS','REF','ALT'])\n",
    "    #df3new = pd.concat([df3, df3FVB_Som]).drop_duplicates(subset=['#CHROM','POS','REF','ALT'], keep=False)\n",
    "    \n",
    "    df4 = pd.merge(df1, df2, on=['#CHROM','POS','REF','ALT'])\n",
    "    df5 = pd.merge(df4, df_Sanger_FVB_Germ, on=['#CHROM','POS','REF','ALT'])\n",
    "    df6 = pd.concat([df4, df5]).drop_duplicates(subset=['#CHROM','POS','REF','ALT'], keep=False)\n",
    "    df7 = pd.merge(df6, df_Haplotype, on=['#CHROM','POS','REF','ALT'])\n",
    "    df = pd.concat([df6, df7]).drop_duplicates(subset=['#CHROM','POS','REF','ALT'], keep=False)\n",
    "    df.dropna(axis='columns', inplace=True)\n",
    "    #df = pd.merge(df3new, df4, on=['#CHROM','POS','REF','ALT'])\n",
    "    df['TYPE'] = df['INFO'].str.extract('ANN=.*?\\|(.*?)\\|')\n",
    "    df['SCORE'] = df['INFO'].str.extract('ANN=.*?\\|.*?\\|(.*?)\\|')\n",
    "    df['GENE'] = df['INFO'].str.extract('ANN=.*?\\|.*?\\|.*?\\|(.*?)\\|')\n",
    "    df['MM#CHROM'] = 'mm' + df['#CHROM'].astype(str)\n",
    "    df['CIRCOS_COLOR'] = 'color=color'+df['SCORE']\n",
    "    #df = df[(df['GENE'] != 'Erbb2')]\n",
    "    df = df.sort_values(by=['somatic_p_value'], ascending=True)\n",
    "    df = df[(df['somatic_p_value'] <= 0.05)]\n",
    "    #df['RANDOM_NUMBER'] = np.random.randint(0,1000,size=(len(df),1))\n",
    "    df['COLOR'] = \"color=color\" + df['REF'] + \"to\" + df['ALT']\n",
    "    #df.dropna(axis='columns', inplace=True)\n",
    "    df = df.sort_values(by=[\"chrom\",\"position\"])\n",
    "    df[\"position\"] = df[\"position\"].astype(float)\n",
    "    df['Intermutation_Distance'] = abs(df['position'] - df['position'].shift(1))\n",
    "    df['Intermutation_Distance_log2'] = np.log2(df['Intermutation_Distance'])\n",
    "    df = df[(df['Intermutation_Distance_log2'] < 24)]\n",
    "    df = df[(df['MM#CHROM'] != 'mmY')]\n",
    "    with pd.ExcelWriter('All_Myc_'+base+'_SNV.xlsx') as writer:\n",
    "        df.to_excel(writer, sheet_name='Total_SNV_Info')\n",
    "        df.to_excel(writer, sheet_name='Circos', columns=['MM#CHROM','POS','POS','Intermutation_Distance_log2','COLOR'])\n",
    "    \n",
    "    \n",
    "    df_SNV = pd.read_excel('All_Myc_'+base+'_SNV.xlsx', sheet_name=\"Circos\", usecols=[1,2,3,4,5])\n",
    "    df_SNV.to_csv('All_'+base+'_Myc_SNV.txt', header=False, index=False, sep='\\t', mode='a')\n",
    "    \n",
    "    df['Chromosome'] = df['#CHROM']\n",
    "    df['Position'] = df['POS']\n",
    "    df['Original Base'] = df['REF']\n",
    "    df['Alternative Base'] = df['ALT']\n",
    "    df['Gene'] = df['GENE']\n",
    "    df['Effect'] = df['TYPE']\n",
    "    df['Predicted Impact'] = df['SCORE']\n",
    "    \n",
    "    with pd.ExcelWriter('All_Myc_'+base+'_SNVs_Pub.xlsx') as writer:\n",
    "        df.to_excel(writer, sheet_name='Total Info', columns=['Chromosome','Position','Original Base','Alternative Base'\\\n",
    "                                                             ,'Gene','Effect','Predicted Impact'])\n",
    "    \n",
    "    \n",
    "    df = df[df['INFO'].str.contains('HIGH|MODERATE|LOW')]\n",
    "    \n",
    "    with pd.ExcelWriter('Impactful_Myc_'+base+'_SNV.xlsx') as writer:\n",
    "        df.to_excel(writer, sheet_name='Total_SNV_Info')\n",
    "        df.to_excel(writer, sheet_name='Circos', columns=['MM#CHROM','POS','POS','CIRCOS_COLOR'])\n",
    "    \n",
    "    with pd.ExcelWriter('Impactful_Myc_'+base+'_SNVs_Pub.xlsx') as writer:\n",
    "        df.to_excel(writer, sheet_name='Total Info', columns=['Chromosome','Position','Original Base','Alternative Base'\\\n",
    "                                                             ,'Gene','Effect','Predicted Impact'])\n",
    "    \n",
    "    SNV_baseList1 = []\n",
    "    SNV_baseList1 = df['GENE']\n",
    "    SNV_baseList1.drop_duplicates(keep='first', inplace=True)\n",
    "    SNV_baseList1.to_csv('Myc_'+base+'_SNV_Gene_List.csv', sep='\\t')\n",
    "    \n",
    "from collections import Counter\n",
    "total_gene_list = []\n",
    "total_gene_list.clear()\n",
    "Myc_List = ['TA39_927','TA41_1938','WT13_842','WT13_864','WT13_1356','WT13_1445','WT13_1576','WT21_222','WT21_1066']\n",
    "#Myc_List = ['3790']\n",
    "for base in Myc_List:\n",
    "# Read in csv files to a dataframe\n",
    "    dfbase = pd.read_csv('Myc_'+base+'_SNV_Gene_List.csv', sep='\\t', usecols=[1], header=None)\n",
    "    base_gene_list = dfbase.values.tolist()\n",
    "    total_gene_list.append(base_gene_list)\n",
    "    flat_total_gene_list = [item for sublist in total_gene_list for item in sublist]\n",
    "    final_flat_total_gene_list = [item for sublist in flat_total_gene_list for item in sublist]\n",
    "albert = Counter(final_flat_total_gene_list)\n",
    "with open('Myc_SNV_Total_Gene_Count.csv','w') as csvfile:\n",
    "    fieldnames=['Gene', 'Count']\n",
    "    writer=csv.writer(csvfile)\n",
    "    writer.writerow(fieldnames)\n",
    "    for key, value in albert.items():\n",
    "        writer.writerow((key, value))\n",
    "print('done')"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 19,
   "id": "a2c914ba",
   "metadata": {},
   "outputs": [
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "C:\\Users\\ANDREC~1\\AppData\\Local\\Temp/ipykernel_26476/3511692749.py:57: SettingWithCopyWarning: \n",
      "A value is trying to be set on a copy of a slice from a DataFrame.\n",
      "Try using .loc[row_indexer,col_indexer] = value instead\n",
      "\n",
      "See the caveats in the documentation: https://pandas.pydata.org/pandas-docs/stable/user_guide/indexing.html#returning-a-view-versus-a-copy\n",
      "  df1['#CHROM'] = df1['chrom']\n",
      "C:\\Users\\ANDREC~1\\AppData\\Local\\Temp/ipykernel_26476/3511692749.py:58: SettingWithCopyWarning: \n",
      "A value is trying to be set on a copy of a slice from a DataFrame.\n",
      "Try using .loc[row_indexer,col_indexer] = value instead\n",
      "\n",
      "See the caveats in the documentation: https://pandas.pydata.org/pandas-docs/stable/user_guide/indexing.html#returning-a-view-versus-a-copy\n",
      "  df1['POS'] = df1['position']\n",
      "C:\\Users\\ANDREC~1\\AppData\\Local\\Temp/ipykernel_26476/3511692749.py:59: SettingWithCopyWarning: \n",
      "A value is trying to be set on a copy of a slice from a DataFrame.\n",
      "Try using .loc[row_indexer,col_indexer] = value instead\n",
      "\n",
      "See the caveats in the documentation: https://pandas.pydata.org/pandas-docs/stable/user_guide/indexing.html#returning-a-view-versus-a-copy\n",
      "  df1['REF'] = df1['ref']\n",
      "C:\\Users\\ANDREC~1\\AppData\\Local\\Temp/ipykernel_26476/3511692749.py:60: SettingWithCopyWarning: \n",
      "A value is trying to be set on a copy of a slice from a DataFrame.\n",
      "Try using .loc[row_indexer,col_indexer] = value instead\n",
      "\n",
      "See the caveats in the documentation: https://pandas.pydata.org/pandas-docs/stable/user_guide/indexing.html#returning-a-view-versus-a-copy\n",
      "  df1['ALT'] = df1['var']\n",
      "C:\\Users\\ANDREC~1\\AppData\\Local\\Temp/ipykernel_26476/3511692749.py:61: SettingWithCopyWarning: \n",
      "A value is trying to be set on a copy of a slice from a DataFrame.\n",
      "Try using .loc[row_indexer,col_indexer] = value instead\n",
      "\n",
      "See the caveats in the documentation: https://pandas.pydata.org/pandas-docs/stable/user_guide/indexing.html#returning-a-view-versus-a-copy\n",
      "  df1['somatic_p_value'] = df1['somatic_p_value'].astype(float)\n"
     ]
    }
   ],
   "source": [
    "#SigProfileExtractor preprocessing\n",
    "# Full Python Script\n",
    "# Python 3.8.8\n",
    "# Pandas 1.2.4\n",
    "# Numpy 1.20.1\n",
    "# Carson Broeker, 10/19/2021\n",
    "#Test script for converting Mutect, SomaticSniper, and Varscan SNV vcf data to a list of genes\n",
    "# First, import necessary modules\n",
    "import math\n",
    "import pandas as pd\n",
    "import numpy as np\n",
    "import sys\n",
    "import csv\n",
    "from functools import reduce\n",
    "\n",
    "# Make it so really large files don't cause the system to error out\n",
    "maxInt = sys.maxsize\n",
    "\n",
    "while True:\n",
    "    # decrease the maxInt value by factor 10 \n",
    "    # as long as the OverflowError occurs.\n",
    "\n",
    "    try:\n",
    "        csv.field_size_limit(maxInt)\n",
    "        break\n",
    "    except OverflowError:\n",
    "        maxInt = int(maxInt/10)\n",
    "# Next, read in all of the VCF files using a for loop\n",
    "# This step also gets rid of superfluous columns for downstream analyses\n",
    "# Modify the List elements to suit your project needs\n",
    "# Make sure all of your vcf files are in the same directory/location as this python script\n",
    "#dfFVB_Var = pd.read_csv(\"FVB_NJVarscanSNVs.vcf.snp\", sep='\\t', usecols=[0,1,2,3,12,14], engine = 'c', dtype=object)\n",
    "#dfFVB_Var = dfFVB_Var[(dfFVB_Var['somatic_status'] == 'Somatic')]\n",
    "#dfFVB_Var['#CHROM'] = dfFVB_Var['chrom']\n",
    "#dfFVB_Var['POS'] = dfFVB_Var['position']\n",
    "#dfFVB_Var['REF'] = dfFVB_Var['ref']\n",
    "#dfFVB_Var['ALT'] = dfFVB_Var['var']\n",
    "\n",
    "#dfFVB_Som = pd.read_csv(\"FVB_NJSomaticSniperCalls.vcf\", sep='\\t', skiprows=17, usecols=[0,1,3,4], engine = 'c', dtype=object)\n",
    "\n",
    "df_Sanger_FVB_Germ = pd.read_csv(\"FVB_NJ.mgp.v5.snps.dbSNP142.vcf\", sep='\\t', skiprows=68, usecols=[0,1,3,4], engine='c', dtype=object)\n",
    "\n",
    "df_Haplotype = pd.read_csv(\"FVB_NJ_HaplotypeCaller.vcf\", sep='\\t', skiprows=93, usecols=[0,1,3,4], engine='c', dtype=object)\n",
    "\n",
    "Myc_List = ['TA39_927','TA41_1938','WT13_842','WT13_864','WT13_1356','WT13_1445','WT13_1576','WT21_222','WT21_1066']\n",
    "#Myc_List = ['WT13_1356']\n",
    "for base in Myc_List:\n",
    "    Varscan_base = pd.read_csv(base+\"VarscanSNVs.vcf.snp\", sep='\\t', usecols=[0,1,2,3,12,14], \\\n",
    "                             engine = 'c', dtype=object)\n",
    "# Must be loaded as engine='c' as the 'python' engine isn't natively equipped with the necessary modules\n",
    "# Next, make each dataframe something easier to read and obvious\n",
    "# If unsure how to modify data, look at the python pandas documentation\n",
    "    df1 = Varscan_base\n",
    "# Use masking to include only columns that have deletions or duplications, no inversions or translocations\n",
    "# Check CNV_Filtering python file for more advanced documentation\n",
    "    df1 = df1[(df1['somatic_status'] == 'Somatic')]\n",
    "    df1['#CHROM'] = df1['chrom']\n",
    "    df1['POS'] = df1['position']\n",
    "    df1['REF'] = df1['ref']\n",
    "    df1['ALT'] = df1['var']\n",
    "    df1['somatic_p_value'] = df1['somatic_p_value'].astype(float)\n",
    "    #df1FVB_Var = pd.merge(df1, dfFVB_Var, on=['#CHROM','POS','REF','ALT'])\n",
    "    #df1new = pd.concat([df1, df1FVB_Var]).drop_duplicates(subset=['#CHROM','POS','REF','ALT'], keep=False)\n",
    "    \n",
    "    Mutect_base = pd.read_csv(base+\"Mutect2Annotated.vcf\", sep='\\t', skiprows=106, usecols=[0,1,3,4,7], \\\n",
    "                             engine = 'c', dtype=object)\n",
    "    df2 = Mutect_base\n",
    "    #df2 = df2[df2['INFO'].str.contains('HIGH|MODERATE|LOW')]\n",
    "    #df2['READ_DEPTH'] = df2['INFO'].str.extract('DP=(\\d+);').astype(int)\n",
    "    \n",
    "    #SomaticSniper_base = pd.read_csv(base+\"_MycSomaticSniperCallsAnnotated.vcf\", sep='\\t', skiprows=22, usecols=[0,1,3,4,7], engine = 'c', dtype=object)\n",
    "    #df3 = SomaticSniper_base\n",
    "    #df3 = df3[df3['INFO'].str.contains('HIGH|MODERATE|LOW')]\n",
    "    #df3FVB_Som = pd.merge(df3, dfFVB_Som, on=['#CHROM','POS','REF','ALT'])\n",
    "    #df3new = pd.concat([df3, df3FVB_Som]).drop_duplicates(subset=['#CHROM','POS','REF','ALT'], keep=False)\n",
    "    \n",
    "    df4 = pd.merge(df1, df2, on=['#CHROM','POS','REF','ALT'])\n",
    "    df5 = pd.merge(df4, df_Sanger_FVB_Germ, on=['#CHROM','POS','REF','ALT'])\n",
    "    df6 = pd.concat([df4, df5]).drop_duplicates(subset=['#CHROM','POS','REF','ALT'], keep=False)\n",
    "    df7 = pd.merge(df6, df_Haplotype, on=['#CHROM','POS','REF','ALT'])\n",
    "    df = pd.concat([df6, df7]).drop_duplicates(subset=['#CHROM','POS','REF','ALT'], keep=False)\n",
    "    df.dropna(axis='columns', inplace=True)\n",
    "    df[\"CHROM\"] = \"chr\" + df[\"#CHROM\"]\n",
    "    #df[\"FILTER\"] = \"PASS\"\n",
    "    df[\"SAMPLE\"] = base\n",
    "    #df = pd.merge(df3new, df4, on=['#CHROM','POS','REF','ALT'])\n",
    "    #df['TYPE'] = df['INFO'].str.extract('ANN=.*?\\|(.*?)\\|')\n",
    "    #df['SCORE'] = df['INFO'].str.extract('ANN=.*?\\|.*?\\|(.*?)\\|')\n",
    "    #df['GENE'] = df['INFO'].str.extract('ANN=.*?\\|.*?\\|.*?\\|(.*?)\\|')\n",
    "    #df['MM#CHROM'] = 'mm' + df['#CHROM'].astype(str)\n",
    "    #df['CIRCOS_COLOR'] = 'color=color'+df['SCORE']\n",
    "    #df = df[(df['GENE'] != 'Erbb2')]\n",
    "    df = df.sort_values(by=['somatic_p_value'], ascending=True)\n",
    "    df = df[(df['somatic_p_value'] <= 0.05)]\n",
    "    #df['RANDOM_NUMBER'] = np.random.randint(0,1000,size=(len(df),1))\n",
    "    #df.dropna(axis='columns', inplace=True)\n",
    "    dfnew = df[[\"SAMPLE\",\"CHROM\",\"POS\",\"REF\",\"ALT\"]].copy()\n",
    "    some_values = [\"chr1\",\"chr2\",\"chr3\",\"chr4\",\"chr5\",\"chr6\",\"chr7\",\"chr8\",\"chr9\",\\\n",
    "              \"chr10\",\"chr11\",\"chr12\",\"chr13\",\"chr14\",\"chr15\",\"chr16\",\"chr17\",\\\n",
    "              \"chr18\",\"chr19\",\"chrX\"]\n",
    "    dfnew = dfnew.loc[df['CHROM'].isin(some_values)]\n",
    "    dfnew = dfnew.sort_values(by=[\"CHROM\",\"POS\"])\n",
    "    dfnew.to_csv(base+\"SigProfileExtractor.vcf\", sep='\\t', index=False)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 6,
   "id": "5fe48dc2",
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "done\n"
     ]
    }
   ],
   "source": [
    "# For venn diagrams\n",
    "from collections import Counter\n",
    "total_gene_list = []\n",
    "total_gene_list.clear()\n",
    "Microacinar_List = ['WT13_842','WT13_864','WT13_1576']\n",
    "Squamous_List = ['TA39_927','WT13_1445','WT21_1066']\n",
    "EMT_List = ['TA41_1938','WT13_1356','WT21_222']\n",
    "#Myc_List = ['3790']\n",
    "total_gene_list.clear()\n",
    "for base in Microacinar_List:\n",
    "# Read in csv files to a dataframe\n",
    "    dfbase = pd.read_csv('Myc_'+base+'_SNV_Gene_List.csv', sep='\\t', usecols=[1], header=None)\n",
    "    base_gene_list = dfbase.values.tolist()\n",
    "    total_gene_list.append(base_gene_list)\n",
    "    flat_total_gene_list = [item for sublist in total_gene_list for item in sublist]\n",
    "    final_flat_total_gene_list = [item for sublist in flat_total_gene_list for item in sublist]\n",
    "albert = Counter(final_flat_total_gene_list)\n",
    "with open('Myc_Microacinar_SNV_Total_Gene_Count.csv','w') as csvfile:\n",
    "    fieldnames=['Gene', 'Count']\n",
    "    writer=csv.writer(csvfile)\n",
    "    writer.writerow(fieldnames)\n",
    "    for key, value in albert.items():\n",
    "        writer.writerow((key, value))\n",
    "total_gene_list.clear()\n",
    "for base in Squamous_List:\n",
    "# Read in csv files to a dataframe\n",
    "    dfbase = pd.read_csv('Myc_'+base+'_SNV_Gene_List.csv', sep='\\t', usecols=[1], header=None)\n",
    "    base_gene_list = dfbase.values.tolist()\n",
    "    total_gene_list.append(base_gene_list)\n",
    "    flat_total_gene_list = [item for sublist in total_gene_list for item in sublist]\n",
    "    final_flat_total_gene_list = [item for sublist in flat_total_gene_list for item in sublist]\n",
    "albert = Counter(final_flat_total_gene_list)\n",
    "with open('Myc_Squamous_SNV_Total_Gene_Count.csv','w') as csvfile:\n",
    "    fieldnames=['Gene', 'Count']\n",
    "    writer=csv.writer(csvfile)\n",
    "    writer.writerow(fieldnames)\n",
    "    for key, value in albert.items():\n",
    "        writer.writerow((key, value))\n",
    "total_gene_list.clear()\n",
    "for base in EMT_List:\n",
    "# Read in csv files to a dataframe\n",
    "    dfbase = pd.read_csv('Myc_'+base+'_SNV_Gene_List.csv', sep='\\t', usecols=[1], header=None)\n",
    "    base_gene_list = dfbase.values.tolist()\n",
    "    total_gene_list.append(base_gene_list)\n",
    "    flat_total_gene_list = [item for sublist in total_gene_list for item in sublist]\n",
    "    final_flat_total_gene_list = [item for sublist in flat_total_gene_list for item in sublist]\n",
    "albert = Counter(final_flat_total_gene_list)\n",
    "with open('Myc_EMT_SNV_Total_Gene_Count.csv','w') as csvfile:\n",
    "    fieldnames=['Gene', 'Count']\n",
    "    writer=csv.writer(csvfile)\n",
    "    writer.writerow(fieldnames)\n",
    "    for key, value in albert.items():\n",
    "        writer.writerow((key, value))\n",
    "print('done')"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 27,
   "id": "858ff4a5",
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "float64\n"
     ]
    },
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>chrom</th>\n",
       "      <th>position</th>\n",
       "      <th>ref</th>\n",
       "      <th>var</th>\n",
       "      <th>somatic_status</th>\n",
       "      <th>somatic_p_value</th>\n",
       "      <th>#CHROM</th>\n",
       "      <th>POS</th>\n",
       "      <th>REF</th>\n",
       "      <th>ALT</th>\n",
       "      <th>INFO</th>\n",
       "      <th>TYPE</th>\n",
       "      <th>SCORE</th>\n",
       "      <th>GENE</th>\n",
       "      <th>MM#CHROM</th>\n",
       "      <th>CIRCOS_COLOR</th>\n",
       "      <th>RANDOM_NUMBER</th>\n",
       "      <th>COLOR</th>\n",
       "      <th>Intermutation_Distance</th>\n",
       "      <th>Intermutation_Distance_log2</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>1</td>\n",
       "      <td>3005150.0</td>\n",
       "      <td>C</td>\n",
       "      <td>A</td>\n",
       "      <td>Somatic</td>\n",
       "      <td>4.953183e-03</td>\n",
       "      <td>1</td>\n",
       "      <td>3005150</td>\n",
       "      <td>C</td>\n",
       "      <td>A</td>\n",
       "      <td>DP=72;ECNT=1;NLOD=11.73;N_ART_LOD=-1.616e+00;P...</td>\n",
       "      <td>intergenic_region</td>\n",
       "      <td>MODIFIER</td>\n",
       "      <td>CHR_START-Xkr4</td>\n",
       "      <td>mm1</td>\n",
       "      <td>color=colorMODIFIER</td>\n",
       "      <td>184</td>\n",
       "      <td>color=colorCtoA</td>\n",
       "      <td>NaN</td>\n",
       "      <td>NaN</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>1</td>\n",
       "      <td>3233691.0</td>\n",
       "      <td>C</td>\n",
       "      <td>A</td>\n",
       "      <td>Somatic</td>\n",
       "      <td>2.015202e-03</td>\n",
       "      <td>1</td>\n",
       "      <td>3233691</td>\n",
       "      <td>C</td>\n",
       "      <td>A</td>\n",
       "      <td>DP=83;ECNT=3;NLOD=16.25;N_ART_LOD=-1.740e+00;P...</td>\n",
       "      <td>intron_variant</td>\n",
       "      <td>MODIFIER</td>\n",
       "      <td>Xkr4</td>\n",
       "      <td>mm1</td>\n",
       "      <td>color=colorMODIFIER</td>\n",
       "      <td>342</td>\n",
       "      <td>color=colorCtoA</td>\n",
       "      <td>228541.0</td>\n",
       "      <td>17.802093</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>1</td>\n",
       "      <td>3233723.0</td>\n",
       "      <td>A</td>\n",
       "      <td>G</td>\n",
       "      <td>Somatic</td>\n",
       "      <td>1.089930e-02</td>\n",
       "      <td>1</td>\n",
       "      <td>3233723</td>\n",
       "      <td>A</td>\n",
       "      <td>G</td>\n",
       "      <td>DP=85;ECNT=3;NLOD=13.50;N_ART_LOD=-1.682e+00;P...</td>\n",
       "      <td>intron_variant</td>\n",
       "      <td>MODIFIER</td>\n",
       "      <td>Xkr4</td>\n",
       "      <td>mm1</td>\n",
       "      <td>color=colorMODIFIER</td>\n",
       "      <td>965</td>\n",
       "      <td>color=colorAtoG</td>\n",
       "      <td>32.0</td>\n",
       "      <td>5.000000</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>1</td>\n",
       "      <td>3235138.0</td>\n",
       "      <td>T</td>\n",
       "      <td>C</td>\n",
       "      <td>Somatic</td>\n",
       "      <td>6.238887e-03</td>\n",
       "      <td>1</td>\n",
       "      <td>3235138</td>\n",
       "      <td>T</td>\n",
       "      <td>C</td>\n",
       "      <td>DP=76;ECNT=1;NLOD=7.86;N_ART_LOD=-1.443e+00;PO...</td>\n",
       "      <td>intron_variant</td>\n",
       "      <td>MODIFIER</td>\n",
       "      <td>Xkr4</td>\n",
       "      <td>mm1</td>\n",
       "      <td>color=colorMODIFIER</td>\n",
       "      <td>508</td>\n",
       "      <td>color=colorTtoC</td>\n",
       "      <td>1415.0</td>\n",
       "      <td>10.466586</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>5</th>\n",
       "      <td>1</td>\n",
       "      <td>3322183.0</td>\n",
       "      <td>T</td>\n",
       "      <td>A</td>\n",
       "      <td>Somatic</td>\n",
       "      <td>3.610201e-05</td>\n",
       "      <td>1</td>\n",
       "      <td>3322183</td>\n",
       "      <td>T</td>\n",
       "      <td>A</td>\n",
       "      <td>DP=91;ECNT=1;NLOD=16.21;N_ART_LOD=-1.750e+00;P...</td>\n",
       "      <td>intron_variant</td>\n",
       "      <td>MODIFIER</td>\n",
       "      <td>Xkr4</td>\n",
       "      <td>mm1</td>\n",
       "      <td>color=colorMODIFIER</td>\n",
       "      <td>370</td>\n",
       "      <td>color=colorTtoA</td>\n",
       "      <td>87045.0</td>\n",
       "      <td>16.409474</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>6</th>\n",
       "      <td>1</td>\n",
       "      <td>3329956.0</td>\n",
       "      <td>A</td>\n",
       "      <td>T</td>\n",
       "      <td>Somatic</td>\n",
       "      <td>2.796281e-03</td>\n",
       "      <td>1</td>\n",
       "      <td>3329956</td>\n",
       "      <td>A</td>\n",
       "      <td>T</td>\n",
       "      <td>DP=77;ECNT=2;NLOD=10.65;N_ART_LOD=-1.594e+00;P...</td>\n",
       "      <td>intron_variant</td>\n",
       "      <td>MODIFIER</td>\n",
       "      <td>Xkr4</td>\n",
       "      <td>mm1</td>\n",
       "      <td>color=colorMODIFIER</td>\n",
       "      <td>456</td>\n",
       "      <td>color=colorAtoT</td>\n",
       "      <td>7773.0</td>\n",
       "      <td>12.924256</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>9</th>\n",
       "      <td>1</td>\n",
       "      <td>3537677.0</td>\n",
       "      <td>C</td>\n",
       "      <td>T</td>\n",
       "      <td>Somatic</td>\n",
       "      <td>1.887913e-03</td>\n",
       "      <td>1</td>\n",
       "      <td>3537677</td>\n",
       "      <td>C</td>\n",
       "      <td>T</td>\n",
       "      <td>DP=82;ECNT=2;NLOD=13.55;N_ART_LOD=-1.663e+00;P...</td>\n",
       "      <td>intron_variant</td>\n",
       "      <td>MODIFIER</td>\n",
       "      <td>Xkr4</td>\n",
       "      <td>mm1</td>\n",
       "      <td>color=colorMODIFIER</td>\n",
       "      <td>215</td>\n",
       "      <td>color=colorCtoT</td>\n",
       "      <td>207721.0</td>\n",
       "      <td>17.664288</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>10</th>\n",
       "      <td>1</td>\n",
       "      <td>3537679.0</td>\n",
       "      <td>C</td>\n",
       "      <td>T</td>\n",
       "      <td>Somatic</td>\n",
       "      <td>9.925075e-04</td>\n",
       "      <td>1</td>\n",
       "      <td>3537679</td>\n",
       "      <td>C</td>\n",
       "      <td>T</td>\n",
       "      <td>DP=82;ECNT=2;NLOD=13.55;N_ART_LOD=-1.663e+00;P...</td>\n",
       "      <td>intron_variant</td>\n",
       "      <td>MODIFIER</td>\n",
       "      <td>Xkr4</td>\n",
       "      <td>mm1</td>\n",
       "      <td>color=colorMODIFIER</td>\n",
       "      <td>17</td>\n",
       "      <td>color=colorCtoT</td>\n",
       "      <td>2.0</td>\n",
       "      <td>1.000000</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>11</th>\n",
       "      <td>1</td>\n",
       "      <td>3537755.0</td>\n",
       "      <td>G</td>\n",
       "      <td>A</td>\n",
       "      <td>Somatic</td>\n",
       "      <td>3.159305e-02</td>\n",
       "      <td>1</td>\n",
       "      <td>3537755</td>\n",
       "      <td>G</td>\n",
       "      <td>A</td>\n",
       "      <td>DP=87;ECNT=4;NLOD=12.94;N_ART_LOD=-1.643e+00;P...</td>\n",
       "      <td>intron_variant</td>\n",
       "      <td>MODIFIER</td>\n",
       "      <td>Xkr4</td>\n",
       "      <td>mm1</td>\n",
       "      <td>color=colorMODIFIER</td>\n",
       "      <td>124</td>\n",
       "      <td>color=colorGtoA</td>\n",
       "      <td>76.0</td>\n",
       "      <td>6.247928</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>12</th>\n",
       "      <td>1</td>\n",
       "      <td>3537779.0</td>\n",
       "      <td>T</td>\n",
       "      <td>A</td>\n",
       "      <td>Somatic</td>\n",
       "      <td>1.334880e-02</td>\n",
       "      <td>1</td>\n",
       "      <td>3537779</td>\n",
       "      <td>T</td>\n",
       "      <td>A</td>\n",
       "      <td>DP=88;ECNT=4;NLOD=13.66;N_ART_LOD=-1.670e+00;P...</td>\n",
       "      <td>intron_variant</td>\n",
       "      <td>MODIFIER</td>\n",
       "      <td>Xkr4</td>\n",
       "      <td>mm1</td>\n",
       "      <td>color=colorMODIFIER</td>\n",
       "      <td>179</td>\n",
       "      <td>color=colorTtoA</td>\n",
       "      <td>24.0</td>\n",
       "      <td>4.584963</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>13</th>\n",
       "      <td>1</td>\n",
       "      <td>3696723.0</td>\n",
       "      <td>A</td>\n",
       "      <td>T</td>\n",
       "      <td>Somatic</td>\n",
       "      <td>1.266498e-03</td>\n",
       "      <td>1</td>\n",
       "      <td>3696723</td>\n",
       "      <td>A</td>\n",
       "      <td>T</td>\n",
       "      <td>DP=67;ECNT=10;NLOD=5.73;N_ART_LOD=1.42;POP_AF=...</td>\n",
       "      <td>intergenic_region</td>\n",
       "      <td>MODIFIER</td>\n",
       "      <td>Xkr4-Rp1</td>\n",
       "      <td>mm1</td>\n",
       "      <td>color=colorMODIFIER</td>\n",
       "      <td>354</td>\n",
       "      <td>color=colorAtoT</td>\n",
       "      <td>158944.0</td>\n",
       "      <td>17.278159</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>14</th>\n",
       "      <td>1</td>\n",
       "      <td>3696729.0</td>\n",
       "      <td>C</td>\n",
       "      <td>T</td>\n",
       "      <td>Somatic</td>\n",
       "      <td>2.131232e-04</td>\n",
       "      <td>1</td>\n",
       "      <td>3696729</td>\n",
       "      <td>C</td>\n",
       "      <td>T</td>\n",
       "      <td>DP=71;ECNT=10;NLOD=11.14;N_ART_LOD=-1.580e+00;...</td>\n",
       "      <td>intergenic_region</td>\n",
       "      <td>MODIFIER</td>\n",
       "      <td>Xkr4-Rp1</td>\n",
       "      <td>mm1</td>\n",
       "      <td>color=colorMODIFIER</td>\n",
       "      <td>549</td>\n",
       "      <td>color=colorCtoT</td>\n",
       "      <td>6.0</td>\n",
       "      <td>2.584963</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>15</th>\n",
       "      <td>1</td>\n",
       "      <td>3696733.0</td>\n",
       "      <td>C</td>\n",
       "      <td>T</td>\n",
       "      <td>Somatic</td>\n",
       "      <td>6.201692e-05</td>\n",
       "      <td>1</td>\n",
       "      <td>3696733</td>\n",
       "      <td>C</td>\n",
       "      <td>T</td>\n",
       "      <td>DP=71;ECNT=10;NLOD=11.44;N_ART_LOD=-1.591e+00;...</td>\n",
       "      <td>intergenic_region</td>\n",
       "      <td>MODIFIER</td>\n",
       "      <td>Xkr4-Rp1</td>\n",
       "      <td>mm1</td>\n",
       "      <td>color=colorMODIFIER</td>\n",
       "      <td>667</td>\n",
       "      <td>color=colorCtoT</td>\n",
       "      <td>4.0</td>\n",
       "      <td>2.000000</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>16</th>\n",
       "      <td>1</td>\n",
       "      <td>3696736.0</td>\n",
       "      <td>G</td>\n",
       "      <td>A</td>\n",
       "      <td>Somatic</td>\n",
       "      <td>9.558999e-06</td>\n",
       "      <td>1</td>\n",
       "      <td>3696736</td>\n",
       "      <td>G</td>\n",
       "      <td>A</td>\n",
       "      <td>DP=67;ECNT=10;NLOD=10.54;N_ART_LOD=-1.556e+00;...</td>\n",
       "      <td>intergenic_region</td>\n",
       "      <td>MODIFIER</td>\n",
       "      <td>Xkr4-Rp1</td>\n",
       "      <td>mm1</td>\n",
       "      <td>color=colorMODIFIER</td>\n",
       "      <td>30</td>\n",
       "      <td>color=colorGtoA</td>\n",
       "      <td>3.0</td>\n",
       "      <td>1.584963</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>17</th>\n",
       "      <td>1</td>\n",
       "      <td>3696740.0</td>\n",
       "      <td>T</td>\n",
       "      <td>C</td>\n",
       "      <td>Somatic</td>\n",
       "      <td>2.869493e-06</td>\n",
       "      <td>1</td>\n",
       "      <td>3696740</td>\n",
       "      <td>T</td>\n",
       "      <td>C</td>\n",
       "      <td>DP=70;ECNT=10;NLOD=10.84;N_ART_LOD=-1.568e+00;...</td>\n",
       "      <td>intergenic_region</td>\n",
       "      <td>MODIFIER</td>\n",
       "      <td>Xkr4-Rp1</td>\n",
       "      <td>mm1</td>\n",
       "      <td>color=colorMODIFIER</td>\n",
       "      <td>116</td>\n",
       "      <td>color=colorTtoC</td>\n",
       "      <td>4.0</td>\n",
       "      <td>2.000000</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>18</th>\n",
       "      <td>1</td>\n",
       "      <td>3696771.0</td>\n",
       "      <td>T</td>\n",
       "      <td>C</td>\n",
       "      <td>Somatic</td>\n",
       "      <td>5.321887e-04</td>\n",
       "      <td>1</td>\n",
       "      <td>3696771</td>\n",
       "      <td>T</td>\n",
       "      <td>C</td>\n",
       "      <td>DP=77;ECNT=10;NLOD=13.00;N_ART_LOD=-1.657e+00;...</td>\n",
       "      <td>intergenic_region</td>\n",
       "      <td>MODIFIER</td>\n",
       "      <td>Xkr4-Rp1</td>\n",
       "      <td>mm1</td>\n",
       "      <td>color=colorMODIFIER</td>\n",
       "      <td>441</td>\n",
       "      <td>color=colorTtoC</td>\n",
       "      <td>31.0</td>\n",
       "      <td>4.954196</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>19</th>\n",
       "      <td>1</td>\n",
       "      <td>3754602.0</td>\n",
       "      <td>T</td>\n",
       "      <td>C</td>\n",
       "      <td>Somatic</td>\n",
       "      <td>2.948005e-02</td>\n",
       "      <td>1</td>\n",
       "      <td>3754602</td>\n",
       "      <td>T</td>\n",
       "      <td>C</td>\n",
       "      <td>DP=36;ECNT=4;NLOD=2.71;N_ART_LOD=-1.049e+00;PO...</td>\n",
       "      <td>intergenic_region</td>\n",
       "      <td>MODIFIER</td>\n",
       "      <td>Xkr4-Rp1</td>\n",
       "      <td>mm1</td>\n",
       "      <td>color=colorMODIFIER</td>\n",
       "      <td>767</td>\n",
       "      <td>color=colorTtoC</td>\n",
       "      <td>57831.0</td>\n",
       "      <td>15.819555</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>21</th>\n",
       "      <td>1</td>\n",
       "      <td>3984985.0</td>\n",
       "      <td>A</td>\n",
       "      <td>G</td>\n",
       "      <td>Somatic</td>\n",
       "      <td>5.912395e-03</td>\n",
       "      <td>1</td>\n",
       "      <td>3984985</td>\n",
       "      <td>A</td>\n",
       "      <td>G</td>\n",
       "      <td>DP=83;ECNT=3;NLOD=15.95;N_ART_LOD=-1.732e+00;P...</td>\n",
       "      <td>intergenic_region</td>\n",
       "      <td>MODIFIER</td>\n",
       "      <td>Xkr4-Rp1</td>\n",
       "      <td>mm1</td>\n",
       "      <td>color=colorMODIFIER</td>\n",
       "      <td>486</td>\n",
       "      <td>color=colorAtoG</td>\n",
       "      <td>230383.0</td>\n",
       "      <td>17.813675</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>22</th>\n",
       "      <td>1</td>\n",
       "      <td>3984993.0</td>\n",
       "      <td>G</td>\n",
       "      <td>A</td>\n",
       "      <td>Somatic</td>\n",
       "      <td>1.435014e-03</td>\n",
       "      <td>1</td>\n",
       "      <td>3984993</td>\n",
       "      <td>G</td>\n",
       "      <td>A</td>\n",
       "      <td>DP=86;ECNT=3;NLOD=16.85;N_ART_LOD=-1.759e+00;P...</td>\n",
       "      <td>intergenic_region</td>\n",
       "      <td>MODIFIER</td>\n",
       "      <td>Xkr4-Rp1</td>\n",
       "      <td>mm1</td>\n",
       "      <td>color=colorMODIFIER</td>\n",
       "      <td>554</td>\n",
       "      <td>color=colorGtoA</td>\n",
       "      <td>8.0</td>\n",
       "      <td>3.000000</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>23</th>\n",
       "      <td>1</td>\n",
       "      <td>3985258.0</td>\n",
       "      <td>T</td>\n",
       "      <td>A</td>\n",
       "      <td>Somatic</td>\n",
       "      <td>2.129642e-09</td>\n",
       "      <td>1</td>\n",
       "      <td>3985258</td>\n",
       "      <td>T</td>\n",
       "      <td>A</td>\n",
       "      <td>DP=134;ECNT=2;NLOD=27.49;N_ART_LOD=-1.985e+00;...</td>\n",
       "      <td>intergenic_region</td>\n",
       "      <td>MODIFIER</td>\n",
       "      <td>Xkr4-Rp1</td>\n",
       "      <td>mm1</td>\n",
       "      <td>color=colorMODIFIER</td>\n",
       "      <td>236</td>\n",
       "      <td>color=colorTtoA</td>\n",
       "      <td>265.0</td>\n",
       "      <td>8.049849</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>24</th>\n",
       "      <td>1</td>\n",
       "      <td>3985291.0</td>\n",
       "      <td>G</td>\n",
       "      <td>A</td>\n",
       "      <td>Somatic</td>\n",
       "      <td>3.009238e-10</td>\n",
       "      <td>1</td>\n",
       "      <td>3985291</td>\n",
       "      <td>G</td>\n",
       "      <td>A</td>\n",
       "      <td>DP=148;ECNT=2;NLOD=25.98;N_ART_LOD=-1.965e+00;...</td>\n",
       "      <td>intergenic_region</td>\n",
       "      <td>MODIFIER</td>\n",
       "      <td>Xkr4-Rp1</td>\n",
       "      <td>mm1</td>\n",
       "      <td>color=colorMODIFIER</td>\n",
       "      <td>3</td>\n",
       "      <td>color=colorGtoA</td>\n",
       "      <td>33.0</td>\n",
       "      <td>5.044394</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>28</th>\n",
       "      <td>1</td>\n",
       "      <td>3987427.0</td>\n",
       "      <td>C</td>\n",
       "      <td>T</td>\n",
       "      <td>Somatic</td>\n",
       "      <td>2.209225e-05</td>\n",
       "      <td>1</td>\n",
       "      <td>3987427</td>\n",
       "      <td>C</td>\n",
       "      <td>T</td>\n",
       "      <td>DP=108;ECNT=4;NLOD=17.36;N_ART_LOD=-1.776e+00;...</td>\n",
       "      <td>intergenic_region</td>\n",
       "      <td>MODIFIER</td>\n",
       "      <td>Xkr4-Rp1</td>\n",
       "      <td>mm1</td>\n",
       "      <td>color=colorMODIFIER</td>\n",
       "      <td>547</td>\n",
       "      <td>color=colorCtoT</td>\n",
       "      <td>2136.0</td>\n",
       "      <td>11.060696</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>29</th>\n",
       "      <td>1</td>\n",
       "      <td>4107012.0</td>\n",
       "      <td>A</td>\n",
       "      <td>G</td>\n",
       "      <td>Somatic</td>\n",
       "      <td>6.406408e-04</td>\n",
       "      <td>1</td>\n",
       "      <td>4107012</td>\n",
       "      <td>A</td>\n",
       "      <td>G</td>\n",
       "      <td>DP=46;ECNT=2;NLOD=4.52;N_ART_LOD=-1.204e+00;PO...</td>\n",
       "      <td>intergenic_region</td>\n",
       "      <td>MODIFIER</td>\n",
       "      <td>Xkr4-Rp1</td>\n",
       "      <td>mm1</td>\n",
       "      <td>color=colorMODIFIER</td>\n",
       "      <td>946</td>\n",
       "      <td>color=colorAtoG</td>\n",
       "      <td>119585.0</td>\n",
       "      <td>16.867677</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>30</th>\n",
       "      <td>1</td>\n",
       "      <td>4398748.0</td>\n",
       "      <td>A</td>\n",
       "      <td>C</td>\n",
       "      <td>Somatic</td>\n",
       "      <td>3.441736e-04</td>\n",
       "      <td>1</td>\n",
       "      <td>4398748</td>\n",
       "      <td>A</td>\n",
       "      <td>C</td>\n",
       "      <td>DP=110;ECNT=3;NLOD=2.37;N_ART_LOD=5.58;POP_AF=...</td>\n",
       "      <td>intron_variant</td>\n",
       "      <td>MODIFIER</td>\n",
       "      <td>Rp1</td>\n",
       "      <td>mm1</td>\n",
       "      <td>color=colorMODIFIER</td>\n",
       "      <td>511</td>\n",
       "      <td>color=colorAtoC</td>\n",
       "      <td>291736.0</td>\n",
       "      <td>18.154304</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>31</th>\n",
       "      <td>1</td>\n",
       "      <td>4399422.0</td>\n",
       "      <td>G</td>\n",
       "      <td>C</td>\n",
       "      <td>Somatic</td>\n",
       "      <td>7.314786e-03</td>\n",
       "      <td>1</td>\n",
       "      <td>4399422</td>\n",
       "      <td>G</td>\n",
       "      <td>C</td>\n",
       "      <td>DP=61;ECNT=6;NLOD=6.32;N_ART_LOD=-1.351e+00;PO...</td>\n",
       "      <td>intron_variant</td>\n",
       "      <td>MODIFIER</td>\n",
       "      <td>Rp1</td>\n",
       "      <td>mm1</td>\n",
       "      <td>color=colorMODIFIER</td>\n",
       "      <td>513</td>\n",
       "      <td>color=colorGtoC</td>\n",
       "      <td>674.0</td>\n",
       "      <td>9.396605</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>33</th>\n",
       "      <td>1</td>\n",
       "      <td>4399436.0</td>\n",
       "      <td>A</td>\n",
       "      <td>G</td>\n",
       "      <td>Somatic</td>\n",
       "      <td>2.387746e-03</td>\n",
       "      <td>1</td>\n",
       "      <td>4399436</td>\n",
       "      <td>A</td>\n",
       "      <td>G</td>\n",
       "      <td>DP=61;ECNT=6;NLOD=5.72;N_ART_LOD=-1.310e+00;PO...</td>\n",
       "      <td>intron_variant</td>\n",
       "      <td>MODIFIER</td>\n",
       "      <td>Rp1</td>\n",
       "      <td>mm1</td>\n",
       "      <td>color=colorMODIFIER</td>\n",
       "      <td>14</td>\n",
       "      <td>color=colorAtoG</td>\n",
       "      <td>14.0</td>\n",
       "      <td>3.807355</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>34</th>\n",
       "      <td>1</td>\n",
       "      <td>4399470.0</td>\n",
       "      <td>T</td>\n",
       "      <td>G</td>\n",
       "      <td>Somatic</td>\n",
       "      <td>1.346642e-03</td>\n",
       "      <td>1</td>\n",
       "      <td>4399470</td>\n",
       "      <td>T</td>\n",
       "      <td>G</td>\n",
       "      <td>DP=69;ECNT=6;NLOD=6.92;N_ART_LOD=-1.409e+00;PO...</td>\n",
       "      <td>intron_variant</td>\n",
       "      <td>MODIFIER</td>\n",
       "      <td>Rp1</td>\n",
       "      <td>mm1</td>\n",
       "      <td>color=colorMODIFIER</td>\n",
       "      <td>852</td>\n",
       "      <td>color=colorTtoG</td>\n",
       "      <td>34.0</td>\n",
       "      <td>5.087463</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>35</th>\n",
       "      <td>1</td>\n",
       "      <td>4399498.0</td>\n",
       "      <td>A</td>\n",
       "      <td>G</td>\n",
       "      <td>Somatic</td>\n",
       "      <td>2.367525e-04</td>\n",
       "      <td>1</td>\n",
       "      <td>4399498</td>\n",
       "      <td>A</td>\n",
       "      <td>G</td>\n",
       "      <td>DP=71;ECNT=6;NLOD=8.73;N_ART_LOD=-1.477e+00;PO...</td>\n",
       "      <td>intron_variant</td>\n",
       "      <td>MODIFIER</td>\n",
       "      <td>Rp1</td>\n",
       "      <td>mm1</td>\n",
       "      <td>color=colorMODIFIER</td>\n",
       "      <td>448</td>\n",
       "      <td>color=colorAtoG</td>\n",
       "      <td>28.0</td>\n",
       "      <td>4.807355</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>36</th>\n",
       "      <td>1</td>\n",
       "      <td>4399506.0</td>\n",
       "      <td>C</td>\n",
       "      <td>T</td>\n",
       "      <td>Somatic</td>\n",
       "      <td>6.806154e-03</td>\n",
       "      <td>1</td>\n",
       "      <td>4399506</td>\n",
       "      <td>C</td>\n",
       "      <td>T</td>\n",
       "      <td>DP=74;ECNT=6;NLOD=9.33;N_ART_LOD=-1.511e+00;PO...</td>\n",
       "      <td>intron_variant</td>\n",
       "      <td>MODIFIER</td>\n",
       "      <td>Rp1</td>\n",
       "      <td>mm1</td>\n",
       "      <td>color=colorMODIFIER</td>\n",
       "      <td>272</td>\n",
       "      <td>color=colorCtoT</td>\n",
       "      <td>8.0</td>\n",
       "      <td>3.000000</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>37</th>\n",
       "      <td>1</td>\n",
       "      <td>4413729.0</td>\n",
       "      <td>G</td>\n",
       "      <td>C</td>\n",
       "      <td>Somatic</td>\n",
       "      <td>8.106748e-04</td>\n",
       "      <td>1</td>\n",
       "      <td>4413729</td>\n",
       "      <td>G</td>\n",
       "      <td>C</td>\n",
       "      <td>DP=34;ECNT=1;NLOD=3.60;N_ART_LOD=-1.140e+00;PO...</td>\n",
       "      <td>upstream_gene_variant</td>\n",
       "      <td>MODIFIER</td>\n",
       "      <td>Rp1</td>\n",
       "      <td>mm1</td>\n",
       "      <td>color=colorMODIFIER</td>\n",
       "      <td>733</td>\n",
       "      <td>color=colorGtoC</td>\n",
       "      <td>14223.0</td>\n",
       "      <td>13.795938</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>38</th>\n",
       "      <td>1</td>\n",
       "      <td>4872780.0</td>\n",
       "      <td>T</td>\n",
       "      <td>A</td>\n",
       "      <td>Somatic</td>\n",
       "      <td>6.513181e-03</td>\n",
       "      <td>1</td>\n",
       "      <td>4872780</td>\n",
       "      <td>T</td>\n",
       "      <td>A</td>\n",
       "      <td>DP=85;ECNT=1;NLOD=10.87;N_ART_LOD=-1.600e+00;P...</td>\n",
       "      <td>intron_variant</td>\n",
       "      <td>MODIFIER</td>\n",
       "      <td>Tcea1</td>\n",
       "      <td>mm1</td>\n",
       "      <td>color=colorMODIFIER</td>\n",
       "      <td>660</td>\n",
       "      <td>color=colorTtoA</td>\n",
       "      <td>459051.0</td>\n",
       "      <td>18.808295</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>39</th>\n",
       "      <td>1</td>\n",
       "      <td>4977086.0</td>\n",
       "      <td>T</td>\n",
       "      <td>C</td>\n",
       "      <td>Somatic</td>\n",
       "      <td>5.472594e-03</td>\n",
       "      <td>1</td>\n",
       "      <td>4977086</td>\n",
       "      <td>T</td>\n",
       "      <td>C</td>\n",
       "      <td>DP=82;ECNT=1;NLOD=6.79;N_ART_LOD=2.02;POP_AF=1...</td>\n",
       "      <td>intron_variant</td>\n",
       "      <td>MODIFIER</td>\n",
       "      <td>Rgs20</td>\n",
       "      <td>mm1</td>\n",
       "      <td>color=colorMODIFIER</td>\n",
       "      <td>335</td>\n",
       "      <td>color=colorTtoC</td>\n",
       "      <td>104306.0</td>\n",
       "      <td>16.670463</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>40</th>\n",
       "      <td>1</td>\n",
       "      <td>4992630.0</td>\n",
       "      <td>G</td>\n",
       "      <td>A</td>\n",
       "      <td>Somatic</td>\n",
       "      <td>2.108984e-04</td>\n",
       "      <td>1</td>\n",
       "      <td>4992630</td>\n",
       "      <td>G</td>\n",
       "      <td>A</td>\n",
       "      <td>DP=95;ECNT=1;NLOD=15.56;N_ART_LOD=-1.737e+00;P...</td>\n",
       "      <td>intron_variant</td>\n",
       "      <td>MODIFIER</td>\n",
       "      <td>Rgs20</td>\n",
       "      <td>mm1</td>\n",
       "      <td>color=colorMODIFIER</td>\n",
       "      <td>725</td>\n",
       "      <td>color=colorGtoA</td>\n",
       "      <td>15544.0</td>\n",
       "      <td>13.924070</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>41</th>\n",
       "      <td>1</td>\n",
       "      <td>5047431.0</td>\n",
       "      <td>A</td>\n",
       "      <td>T</td>\n",
       "      <td>Somatic</td>\n",
       "      <td>2.991320e-04</td>\n",
       "      <td>1</td>\n",
       "      <td>5047431</td>\n",
       "      <td>A</td>\n",
       "      <td>T</td>\n",
       "      <td>DP=66;ECNT=2;NLOD=6.87;N_ART_LOD=-1.387e+00;PO...</td>\n",
       "      <td>intron_variant</td>\n",
       "      <td>MODIFIER</td>\n",
       "      <td>Rgs20</td>\n",
       "      <td>mm1</td>\n",
       "      <td>color=colorMODIFIER</td>\n",
       "      <td>178</td>\n",
       "      <td>color=colorAtoT</td>\n",
       "      <td>54801.0</td>\n",
       "      <td>15.741915</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>42</th>\n",
       "      <td>1</td>\n",
       "      <td>5166692.0</td>\n",
       "      <td>T</td>\n",
       "      <td>C</td>\n",
       "      <td>Somatic</td>\n",
       "      <td>1.391320e-02</td>\n",
       "      <td>1</td>\n",
       "      <td>5166692</td>\n",
       "      <td>T</td>\n",
       "      <td>C</td>\n",
       "      <td>DP=51;ECNT=7;NLOD=9.93;N_ART_LOD=-1.537e+00;PO...</td>\n",
       "      <td>downstream_gene_variant</td>\n",
       "      <td>MODIFIER</td>\n",
       "      <td>Atp6v1h</td>\n",
       "      <td>mm1</td>\n",
       "      <td>color=colorMODIFIER</td>\n",
       "      <td>362</td>\n",
       "      <td>color=colorTtoC</td>\n",
       "      <td>119261.0</td>\n",
       "      <td>16.863763</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>43</th>\n",
       "      <td>1</td>\n",
       "      <td>5261610.0</td>\n",
       "      <td>T</td>\n",
       "      <td>C</td>\n",
       "      <td>Somatic</td>\n",
       "      <td>6.568187e-06</td>\n",
       "      <td>1</td>\n",
       "      <td>5261610</td>\n",
       "      <td>T</td>\n",
       "      <td>C</td>\n",
       "      <td>DP=95;ECNT=1;NLOD=10.50;N_ART_LOD=-1.631e+00;P...</td>\n",
       "      <td>intergenic_region</td>\n",
       "      <td>MODIFIER</td>\n",
       "      <td>Atp6v1h-Oprk1</td>\n",
       "      <td>mm1</td>\n",
       "      <td>color=colorMODIFIER</td>\n",
       "      <td>228</td>\n",
       "      <td>color=colorTtoC</td>\n",
       "      <td>94918.0</td>\n",
       "      <td>16.534394</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>44</th>\n",
       "      <td>1</td>\n",
       "      <td>5327817.0</td>\n",
       "      <td>C</td>\n",
       "      <td>T</td>\n",
       "      <td>Somatic</td>\n",
       "      <td>1.792022e-03</td>\n",
       "      <td>1</td>\n",
       "      <td>5327817</td>\n",
       "      <td>C</td>\n",
       "      <td>T</td>\n",
       "      <td>DP=122;ECNT=8;NLOD=23.39;N_ART_LOD=-1.897e+00;...</td>\n",
       "      <td>intergenic_region</td>\n",
       "      <td>MODIFIER</td>\n",
       "      <td>Atp6v1h-Oprk1</td>\n",
       "      <td>mm1</td>\n",
       "      <td>color=colorMODIFIER</td>\n",
       "      <td>5</td>\n",
       "      <td>color=colorCtoT</td>\n",
       "      <td>66207.0</td>\n",
       "      <td>16.014696</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>45</th>\n",
       "      <td>1</td>\n",
       "      <td>5327826.0</td>\n",
       "      <td>C</td>\n",
       "      <td>T</td>\n",
       "      <td>Somatic</td>\n",
       "      <td>1.917423e-03</td>\n",
       "      <td>1</td>\n",
       "      <td>5327826</td>\n",
       "      <td>C</td>\n",
       "      <td>T</td>\n",
       "      <td>DP=111;ECNT=8;NLOD=20.17;N_ART_LOD=-1.832e+00;...</td>\n",
       "      <td>intergenic_region</td>\n",
       "      <td>MODIFIER</td>\n",
       "      <td>Atp6v1h-Oprk1</td>\n",
       "      <td>mm1</td>\n",
       "      <td>color=colorMODIFIER</td>\n",
       "      <td>932</td>\n",
       "      <td>color=colorCtoT</td>\n",
       "      <td>9.0</td>\n",
       "      <td>3.169925</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>46</th>\n",
       "      <td>1</td>\n",
       "      <td>5327832.0</td>\n",
       "      <td>G</td>\n",
       "      <td>C</td>\n",
       "      <td>Somatic</td>\n",
       "      <td>3.291304e-03</td>\n",
       "      <td>1</td>\n",
       "      <td>5327832</td>\n",
       "      <td>G</td>\n",
       "      <td>C</td>\n",
       "      <td>DP=106;ECNT=8;NLOD=18.96;N_ART_LOD=-1.806e+00;...</td>\n",
       "      <td>intergenic_region</td>\n",
       "      <td>MODIFIER</td>\n",
       "      <td>Atp6v1h-Oprk1</td>\n",
       "      <td>mm1</td>\n",
       "      <td>color=colorMODIFIER</td>\n",
       "      <td>909</td>\n",
       "      <td>color=colorGtoC</td>\n",
       "      <td>6.0</td>\n",
       "      <td>2.584963</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>47</th>\n",
       "      <td>1</td>\n",
       "      <td>5327834.0</td>\n",
       "      <td>T</td>\n",
       "      <td>C</td>\n",
       "      <td>Somatic</td>\n",
       "      <td>3.622641e-05</td>\n",
       "      <td>1</td>\n",
       "      <td>5327834</td>\n",
       "      <td>T</td>\n",
       "      <td>C</td>\n",
       "      <td>DP=104;ECNT=8;NLOD=18.06;N_ART_LOD=-1.788e+00;...</td>\n",
       "      <td>intergenic_region</td>\n",
       "      <td>MODIFIER</td>\n",
       "      <td>Atp6v1h-Oprk1</td>\n",
       "      <td>mm1</td>\n",
       "      <td>color=colorMODIFIER</td>\n",
       "      <td>526</td>\n",
       "      <td>color=colorTtoC</td>\n",
       "      <td>2.0</td>\n",
       "      <td>1.000000</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>48</th>\n",
       "      <td>1</td>\n",
       "      <td>5327866.0</td>\n",
       "      <td>C</td>\n",
       "      <td>A</td>\n",
       "      <td>Somatic</td>\n",
       "      <td>1.337213e-04</td>\n",
       "      <td>1</td>\n",
       "      <td>5327866</td>\n",
       "      <td>C</td>\n",
       "      <td>A</td>\n",
       "      <td>DP=74;ECNT=8;NLOD=8.03;N_ART_LOD=-1.476e+00;PO...</td>\n",
       "      <td>intergenic_region</td>\n",
       "      <td>MODIFIER</td>\n",
       "      <td>Atp6v1h-Oprk1</td>\n",
       "      <td>mm1</td>\n",
       "      <td>color=colorMODIFIER</td>\n",
       "      <td>56</td>\n",
       "      <td>color=colorCtoA</td>\n",
       "      <td>32.0</td>\n",
       "      <td>5.000000</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>49</th>\n",
       "      <td>1</td>\n",
       "      <td>5358007.0</td>\n",
       "      <td>A</td>\n",
       "      <td>G</td>\n",
       "      <td>Somatic</td>\n",
       "      <td>1.518185e-05</td>\n",
       "      <td>1</td>\n",
       "      <td>5358007</td>\n",
       "      <td>A</td>\n",
       "      <td>G</td>\n",
       "      <td>DP=91;ECNT=1;NLOD=14.74;N_ART_LOD=-1.703e+00;P...</td>\n",
       "      <td>intergenic_region</td>\n",
       "      <td>MODIFIER</td>\n",
       "      <td>Atp6v1h-Oprk1</td>\n",
       "      <td>mm1</td>\n",
       "      <td>color=colorMODIFIER</td>\n",
       "      <td>997</td>\n",
       "      <td>color=colorAtoG</td>\n",
       "      <td>30141.0</td>\n",
       "      <td>14.879440</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>50</th>\n",
       "      <td>1</td>\n",
       "      <td>5725373.0</td>\n",
       "      <td>C</td>\n",
       "      <td>G</td>\n",
       "      <td>Somatic</td>\n",
       "      <td>1.928490e-12</td>\n",
       "      <td>1</td>\n",
       "      <td>5725373</td>\n",
       "      <td>C</td>\n",
       "      <td>G</td>\n",
       "      <td>DP=101;ECNT=1;NLOD=15.65;N_ART_LOD=-1.745e+00;...</td>\n",
       "      <td>intergenic_region</td>\n",
       "      <td>MODIFIER</td>\n",
       "      <td>Oprk1-Npbwr1</td>\n",
       "      <td>mm1</td>\n",
       "      <td>color=colorMODIFIER</td>\n",
       "      <td>34</td>\n",
       "      <td>color=colorCtoG</td>\n",
       "      <td>367366.0</td>\n",
       "      <td>18.486859</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>51</th>\n",
       "      <td>1</td>\n",
       "      <td>5750497.0</td>\n",
       "      <td>T</td>\n",
       "      <td>C</td>\n",
       "      <td>Somatic</td>\n",
       "      <td>1.752602e-04</td>\n",
       "      <td>1</td>\n",
       "      <td>5750497</td>\n",
       "      <td>T</td>\n",
       "      <td>C</td>\n",
       "      <td>DP=64;ECNT=6;NLOD=9.32;N_ART_LOD=-1.511e+00;PO...</td>\n",
       "      <td>intergenic_region</td>\n",
       "      <td>MODIFIER</td>\n",
       "      <td>Oprk1-Npbwr1</td>\n",
       "      <td>mm1</td>\n",
       "      <td>color=colorMODIFIER</td>\n",
       "      <td>502</td>\n",
       "      <td>color=colorTtoC</td>\n",
       "      <td>25124.0</td>\n",
       "      <td>14.616779</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>52</th>\n",
       "      <td>1</td>\n",
       "      <td>5750501.0</td>\n",
       "      <td>T</td>\n",
       "      <td>C</td>\n",
       "      <td>Somatic</td>\n",
       "      <td>1.384281e-03</td>\n",
       "      <td>1</td>\n",
       "      <td>5750501</td>\n",
       "      <td>T</td>\n",
       "      <td>C</td>\n",
       "      <td>DP=65;ECNT=6;NLOD=9.93;N_ART_LOD=-1.531e+00;PO...</td>\n",
       "      <td>intergenic_region</td>\n",
       "      <td>MODIFIER</td>\n",
       "      <td>Oprk1-Npbwr1</td>\n",
       "      <td>mm1</td>\n",
       "      <td>color=colorMODIFIER</td>\n",
       "      <td>995</td>\n",
       "      <td>color=colorTtoC</td>\n",
       "      <td>4.0</td>\n",
       "      <td>2.000000</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>53</th>\n",
       "      <td>1</td>\n",
       "      <td>5750506.0</td>\n",
       "      <td>A</td>\n",
       "      <td>C</td>\n",
       "      <td>Somatic</td>\n",
       "      <td>2.285524e-06</td>\n",
       "      <td>1</td>\n",
       "      <td>5750506</td>\n",
       "      <td>A</td>\n",
       "      <td>C</td>\n",
       "      <td>DP=56;ECNT=6;NLOD=9.63;N_ART_LOD=-1.519e+00;PO...</td>\n",
       "      <td>intergenic_region</td>\n",
       "      <td>MODIFIER</td>\n",
       "      <td>Oprk1-Npbwr1</td>\n",
       "      <td>mm1</td>\n",
       "      <td>color=colorMODIFIER</td>\n",
       "      <td>336</td>\n",
       "      <td>color=colorAtoC</td>\n",
       "      <td>5.0</td>\n",
       "      <td>2.321928</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>54</th>\n",
       "      <td>1</td>\n",
       "      <td>5941829.0</td>\n",
       "      <td>C</td>\n",
       "      <td>G</td>\n",
       "      <td>Somatic</td>\n",
       "      <td>5.689900e-03</td>\n",
       "      <td>1</td>\n",
       "      <td>5941829</td>\n",
       "      <td>C</td>\n",
       "      <td>G</td>\n",
       "      <td>DP=37;ECNT=2;NLOD=3.61;N_ART_LOD=-1.128e+00;PO...</td>\n",
       "      <td>intergenic_region</td>\n",
       "      <td>MODIFIER</td>\n",
       "      <td>Npbwr1-Rb1cc1</td>\n",
       "      <td>mm1</td>\n",
       "      <td>color=colorMODIFIER</td>\n",
       "      <td>969</td>\n",
       "      <td>color=colorCtoG</td>\n",
       "      <td>191323.0</td>\n",
       "      <td>17.545651</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>55</th>\n",
       "      <td>1</td>\n",
       "      <td>6380073.0</td>\n",
       "      <td>G</td>\n",
       "      <td>A</td>\n",
       "      <td>Somatic</td>\n",
       "      <td>3.978947e-03</td>\n",
       "      <td>1</td>\n",
       "      <td>6380073</td>\n",
       "      <td>G</td>\n",
       "      <td>A</td>\n",
       "      <td>DP=74;ECNT=1;NLOD=11.13;N_ART_LOD=-1.594e+00;P...</td>\n",
       "      <td>intron_variant</td>\n",
       "      <td>MODIFIER</td>\n",
       "      <td>Fam150a</td>\n",
       "      <td>mm1</td>\n",
       "      <td>color=colorMODIFIER</td>\n",
       "      <td>98</td>\n",
       "      <td>color=colorGtoA</td>\n",
       "      <td>438244.0</td>\n",
       "      <td>18.741375</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>56</th>\n",
       "      <td>1</td>\n",
       "      <td>6429116.0</td>\n",
       "      <td>C</td>\n",
       "      <td>T</td>\n",
       "      <td>Somatic</td>\n",
       "      <td>1.221814e-02</td>\n",
       "      <td>1</td>\n",
       "      <td>6429116</td>\n",
       "      <td>C</td>\n",
       "      <td>T</td>\n",
       "      <td>DP=62;ECNT=1;NLOD=6.99;N_ART_LOD=-1.353e+00;PO...</td>\n",
       "      <td>intergenic_region</td>\n",
       "      <td>MODIFIER</td>\n",
       "      <td>Fam150a-St18</td>\n",
       "      <td>mm1</td>\n",
       "      <td>color=colorMODIFIER</td>\n",
       "      <td>495</td>\n",
       "      <td>color=colorCtoT</td>\n",
       "      <td>49043.0</td>\n",
       "      <td>15.581760</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>57</th>\n",
       "      <td>1</td>\n",
       "      <td>6518121.0</td>\n",
       "      <td>C</td>\n",
       "      <td>T</td>\n",
       "      <td>Somatic</td>\n",
       "      <td>1.067149e-03</td>\n",
       "      <td>1</td>\n",
       "      <td>6518121</td>\n",
       "      <td>C</td>\n",
       "      <td>T</td>\n",
       "      <td>DP=84;ECNT=2;NLOD=15.96;N_ART_LOD=0.830;POP_AF...</td>\n",
       "      <td>intron_variant</td>\n",
       "      <td>MODIFIER</td>\n",
       "      <td>St18</td>\n",
       "      <td>mm1</td>\n",
       "      <td>color=colorMODIFIER</td>\n",
       "      <td>873</td>\n",
       "      <td>color=colorCtoT</td>\n",
       "      <td>89005.0</td>\n",
       "      <td>16.441599</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "   chrom   position ref var somatic_status  somatic_p_value #CHROM      POS  \\\n",
       "0      1  3005150.0   C   A        Somatic     4.953183e-03      1  3005150   \n",
       "2      1  3233691.0   C   A        Somatic     2.015202e-03      1  3233691   \n",
       "3      1  3233723.0   A   G        Somatic     1.089930e-02      1  3233723   \n",
       "4      1  3235138.0   T   C        Somatic     6.238887e-03      1  3235138   \n",
       "5      1  3322183.0   T   A        Somatic     3.610201e-05      1  3322183   \n",
       "6      1  3329956.0   A   T        Somatic     2.796281e-03      1  3329956   \n",
       "9      1  3537677.0   C   T        Somatic     1.887913e-03      1  3537677   \n",
       "10     1  3537679.0   C   T        Somatic     9.925075e-04      1  3537679   \n",
       "11     1  3537755.0   G   A        Somatic     3.159305e-02      1  3537755   \n",
       "12     1  3537779.0   T   A        Somatic     1.334880e-02      1  3537779   \n",
       "13     1  3696723.0   A   T        Somatic     1.266498e-03      1  3696723   \n",
       "14     1  3696729.0   C   T        Somatic     2.131232e-04      1  3696729   \n",
       "15     1  3696733.0   C   T        Somatic     6.201692e-05      1  3696733   \n",
       "16     1  3696736.0   G   A        Somatic     9.558999e-06      1  3696736   \n",
       "17     1  3696740.0   T   C        Somatic     2.869493e-06      1  3696740   \n",
       "18     1  3696771.0   T   C        Somatic     5.321887e-04      1  3696771   \n",
       "19     1  3754602.0   T   C        Somatic     2.948005e-02      1  3754602   \n",
       "21     1  3984985.0   A   G        Somatic     5.912395e-03      1  3984985   \n",
       "22     1  3984993.0   G   A        Somatic     1.435014e-03      1  3984993   \n",
       "23     1  3985258.0   T   A        Somatic     2.129642e-09      1  3985258   \n",
       "24     1  3985291.0   G   A        Somatic     3.009238e-10      1  3985291   \n",
       "28     1  3987427.0   C   T        Somatic     2.209225e-05      1  3987427   \n",
       "29     1  4107012.0   A   G        Somatic     6.406408e-04      1  4107012   \n",
       "30     1  4398748.0   A   C        Somatic     3.441736e-04      1  4398748   \n",
       "31     1  4399422.0   G   C        Somatic     7.314786e-03      1  4399422   \n",
       "33     1  4399436.0   A   G        Somatic     2.387746e-03      1  4399436   \n",
       "34     1  4399470.0   T   G        Somatic     1.346642e-03      1  4399470   \n",
       "35     1  4399498.0   A   G        Somatic     2.367525e-04      1  4399498   \n",
       "36     1  4399506.0   C   T        Somatic     6.806154e-03      1  4399506   \n",
       "37     1  4413729.0   G   C        Somatic     8.106748e-04      1  4413729   \n",
       "38     1  4872780.0   T   A        Somatic     6.513181e-03      1  4872780   \n",
       "39     1  4977086.0   T   C        Somatic     5.472594e-03      1  4977086   \n",
       "40     1  4992630.0   G   A        Somatic     2.108984e-04      1  4992630   \n",
       "41     1  5047431.0   A   T        Somatic     2.991320e-04      1  5047431   \n",
       "42     1  5166692.0   T   C        Somatic     1.391320e-02      1  5166692   \n",
       "43     1  5261610.0   T   C        Somatic     6.568187e-06      1  5261610   \n",
       "44     1  5327817.0   C   T        Somatic     1.792022e-03      1  5327817   \n",
       "45     1  5327826.0   C   T        Somatic     1.917423e-03      1  5327826   \n",
       "46     1  5327832.0   G   C        Somatic     3.291304e-03      1  5327832   \n",
       "47     1  5327834.0   T   C        Somatic     3.622641e-05      1  5327834   \n",
       "48     1  5327866.0   C   A        Somatic     1.337213e-04      1  5327866   \n",
       "49     1  5358007.0   A   G        Somatic     1.518185e-05      1  5358007   \n",
       "50     1  5725373.0   C   G        Somatic     1.928490e-12      1  5725373   \n",
       "51     1  5750497.0   T   C        Somatic     1.752602e-04      1  5750497   \n",
       "52     1  5750501.0   T   C        Somatic     1.384281e-03      1  5750501   \n",
       "53     1  5750506.0   A   C        Somatic     2.285524e-06      1  5750506   \n",
       "54     1  5941829.0   C   G        Somatic     5.689900e-03      1  5941829   \n",
       "55     1  6380073.0   G   A        Somatic     3.978947e-03      1  6380073   \n",
       "56     1  6429116.0   C   T        Somatic     1.221814e-02      1  6429116   \n",
       "57     1  6518121.0   C   T        Somatic     1.067149e-03      1  6518121   \n",
       "\n",
       "   REF ALT                                               INFO  \\\n",
       "0    C   A  DP=72;ECNT=1;NLOD=11.73;N_ART_LOD=-1.616e+00;P...   \n",
       "2    C   A  DP=83;ECNT=3;NLOD=16.25;N_ART_LOD=-1.740e+00;P...   \n",
       "3    A   G  DP=85;ECNT=3;NLOD=13.50;N_ART_LOD=-1.682e+00;P...   \n",
       "4    T   C  DP=76;ECNT=1;NLOD=7.86;N_ART_LOD=-1.443e+00;PO...   \n",
       "5    T   A  DP=91;ECNT=1;NLOD=16.21;N_ART_LOD=-1.750e+00;P...   \n",
       "6    A   T  DP=77;ECNT=2;NLOD=10.65;N_ART_LOD=-1.594e+00;P...   \n",
       "9    C   T  DP=82;ECNT=2;NLOD=13.55;N_ART_LOD=-1.663e+00;P...   \n",
       "10   C   T  DP=82;ECNT=2;NLOD=13.55;N_ART_LOD=-1.663e+00;P...   \n",
       "11   G   A  DP=87;ECNT=4;NLOD=12.94;N_ART_LOD=-1.643e+00;P...   \n",
       "12   T   A  DP=88;ECNT=4;NLOD=13.66;N_ART_LOD=-1.670e+00;P...   \n",
       "13   A   T  DP=67;ECNT=10;NLOD=5.73;N_ART_LOD=1.42;POP_AF=...   \n",
       "14   C   T  DP=71;ECNT=10;NLOD=11.14;N_ART_LOD=-1.580e+00;...   \n",
       "15   C   T  DP=71;ECNT=10;NLOD=11.44;N_ART_LOD=-1.591e+00;...   \n",
       "16   G   A  DP=67;ECNT=10;NLOD=10.54;N_ART_LOD=-1.556e+00;...   \n",
       "17   T   C  DP=70;ECNT=10;NLOD=10.84;N_ART_LOD=-1.568e+00;...   \n",
       "18   T   C  DP=77;ECNT=10;NLOD=13.00;N_ART_LOD=-1.657e+00;...   \n",
       "19   T   C  DP=36;ECNT=4;NLOD=2.71;N_ART_LOD=-1.049e+00;PO...   \n",
       "21   A   G  DP=83;ECNT=3;NLOD=15.95;N_ART_LOD=-1.732e+00;P...   \n",
       "22   G   A  DP=86;ECNT=3;NLOD=16.85;N_ART_LOD=-1.759e+00;P...   \n",
       "23   T   A  DP=134;ECNT=2;NLOD=27.49;N_ART_LOD=-1.985e+00;...   \n",
       "24   G   A  DP=148;ECNT=2;NLOD=25.98;N_ART_LOD=-1.965e+00;...   \n",
       "28   C   T  DP=108;ECNT=4;NLOD=17.36;N_ART_LOD=-1.776e+00;...   \n",
       "29   A   G  DP=46;ECNT=2;NLOD=4.52;N_ART_LOD=-1.204e+00;PO...   \n",
       "30   A   C  DP=110;ECNT=3;NLOD=2.37;N_ART_LOD=5.58;POP_AF=...   \n",
       "31   G   C  DP=61;ECNT=6;NLOD=6.32;N_ART_LOD=-1.351e+00;PO...   \n",
       "33   A   G  DP=61;ECNT=6;NLOD=5.72;N_ART_LOD=-1.310e+00;PO...   \n",
       "34   T   G  DP=69;ECNT=6;NLOD=6.92;N_ART_LOD=-1.409e+00;PO...   \n",
       "35   A   G  DP=71;ECNT=6;NLOD=8.73;N_ART_LOD=-1.477e+00;PO...   \n",
       "36   C   T  DP=74;ECNT=6;NLOD=9.33;N_ART_LOD=-1.511e+00;PO...   \n",
       "37   G   C  DP=34;ECNT=1;NLOD=3.60;N_ART_LOD=-1.140e+00;PO...   \n",
       "38   T   A  DP=85;ECNT=1;NLOD=10.87;N_ART_LOD=-1.600e+00;P...   \n",
       "39   T   C  DP=82;ECNT=1;NLOD=6.79;N_ART_LOD=2.02;POP_AF=1...   \n",
       "40   G   A  DP=95;ECNT=1;NLOD=15.56;N_ART_LOD=-1.737e+00;P...   \n",
       "41   A   T  DP=66;ECNT=2;NLOD=6.87;N_ART_LOD=-1.387e+00;PO...   \n",
       "42   T   C  DP=51;ECNT=7;NLOD=9.93;N_ART_LOD=-1.537e+00;PO...   \n",
       "43   T   C  DP=95;ECNT=1;NLOD=10.50;N_ART_LOD=-1.631e+00;P...   \n",
       "44   C   T  DP=122;ECNT=8;NLOD=23.39;N_ART_LOD=-1.897e+00;...   \n",
       "45   C   T  DP=111;ECNT=8;NLOD=20.17;N_ART_LOD=-1.832e+00;...   \n",
       "46   G   C  DP=106;ECNT=8;NLOD=18.96;N_ART_LOD=-1.806e+00;...   \n",
       "47   T   C  DP=104;ECNT=8;NLOD=18.06;N_ART_LOD=-1.788e+00;...   \n",
       "48   C   A  DP=74;ECNT=8;NLOD=8.03;N_ART_LOD=-1.476e+00;PO...   \n",
       "49   A   G  DP=91;ECNT=1;NLOD=14.74;N_ART_LOD=-1.703e+00;P...   \n",
       "50   C   G  DP=101;ECNT=1;NLOD=15.65;N_ART_LOD=-1.745e+00;...   \n",
       "51   T   C  DP=64;ECNT=6;NLOD=9.32;N_ART_LOD=-1.511e+00;PO...   \n",
       "52   T   C  DP=65;ECNT=6;NLOD=9.93;N_ART_LOD=-1.531e+00;PO...   \n",
       "53   A   C  DP=56;ECNT=6;NLOD=9.63;N_ART_LOD=-1.519e+00;PO...   \n",
       "54   C   G  DP=37;ECNT=2;NLOD=3.61;N_ART_LOD=-1.128e+00;PO...   \n",
       "55   G   A  DP=74;ECNT=1;NLOD=11.13;N_ART_LOD=-1.594e+00;P...   \n",
       "56   C   T  DP=62;ECNT=1;NLOD=6.99;N_ART_LOD=-1.353e+00;PO...   \n",
       "57   C   T  DP=84;ECNT=2;NLOD=15.96;N_ART_LOD=0.830;POP_AF...   \n",
       "\n",
       "                       TYPE     SCORE            GENE MM#CHROM  \\\n",
       "0         intergenic_region  MODIFIER  CHR_START-Xkr4      mm1   \n",
       "2            intron_variant  MODIFIER            Xkr4      mm1   \n",
       "3            intron_variant  MODIFIER            Xkr4      mm1   \n",
       "4            intron_variant  MODIFIER            Xkr4      mm1   \n",
       "5            intron_variant  MODIFIER            Xkr4      mm1   \n",
       "6            intron_variant  MODIFIER            Xkr4      mm1   \n",
       "9            intron_variant  MODIFIER            Xkr4      mm1   \n",
       "10           intron_variant  MODIFIER            Xkr4      mm1   \n",
       "11           intron_variant  MODIFIER            Xkr4      mm1   \n",
       "12           intron_variant  MODIFIER            Xkr4      mm1   \n",
       "13        intergenic_region  MODIFIER        Xkr4-Rp1      mm1   \n",
       "14        intergenic_region  MODIFIER        Xkr4-Rp1      mm1   \n",
       "15        intergenic_region  MODIFIER        Xkr4-Rp1      mm1   \n",
       "16        intergenic_region  MODIFIER        Xkr4-Rp1      mm1   \n",
       "17        intergenic_region  MODIFIER        Xkr4-Rp1      mm1   \n",
       "18        intergenic_region  MODIFIER        Xkr4-Rp1      mm1   \n",
       "19        intergenic_region  MODIFIER        Xkr4-Rp1      mm1   \n",
       "21        intergenic_region  MODIFIER        Xkr4-Rp1      mm1   \n",
       "22        intergenic_region  MODIFIER        Xkr4-Rp1      mm1   \n",
       "23        intergenic_region  MODIFIER        Xkr4-Rp1      mm1   \n",
       "24        intergenic_region  MODIFIER        Xkr4-Rp1      mm1   \n",
       "28        intergenic_region  MODIFIER        Xkr4-Rp1      mm1   \n",
       "29        intergenic_region  MODIFIER        Xkr4-Rp1      mm1   \n",
       "30           intron_variant  MODIFIER             Rp1      mm1   \n",
       "31           intron_variant  MODIFIER             Rp1      mm1   \n",
       "33           intron_variant  MODIFIER             Rp1      mm1   \n",
       "34           intron_variant  MODIFIER             Rp1      mm1   \n",
       "35           intron_variant  MODIFIER             Rp1      mm1   \n",
       "36           intron_variant  MODIFIER             Rp1      mm1   \n",
       "37    upstream_gene_variant  MODIFIER             Rp1      mm1   \n",
       "38           intron_variant  MODIFIER           Tcea1      mm1   \n",
       "39           intron_variant  MODIFIER           Rgs20      mm1   \n",
       "40           intron_variant  MODIFIER           Rgs20      mm1   \n",
       "41           intron_variant  MODIFIER           Rgs20      mm1   \n",
       "42  downstream_gene_variant  MODIFIER         Atp6v1h      mm1   \n",
       "43        intergenic_region  MODIFIER   Atp6v1h-Oprk1      mm1   \n",
       "44        intergenic_region  MODIFIER   Atp6v1h-Oprk1      mm1   \n",
       "45        intergenic_region  MODIFIER   Atp6v1h-Oprk1      mm1   \n",
       "46        intergenic_region  MODIFIER   Atp6v1h-Oprk1      mm1   \n",
       "47        intergenic_region  MODIFIER   Atp6v1h-Oprk1      mm1   \n",
       "48        intergenic_region  MODIFIER   Atp6v1h-Oprk1      mm1   \n",
       "49        intergenic_region  MODIFIER   Atp6v1h-Oprk1      mm1   \n",
       "50        intergenic_region  MODIFIER    Oprk1-Npbwr1      mm1   \n",
       "51        intergenic_region  MODIFIER    Oprk1-Npbwr1      mm1   \n",
       "52        intergenic_region  MODIFIER    Oprk1-Npbwr1      mm1   \n",
       "53        intergenic_region  MODIFIER    Oprk1-Npbwr1      mm1   \n",
       "54        intergenic_region  MODIFIER   Npbwr1-Rb1cc1      mm1   \n",
       "55           intron_variant  MODIFIER         Fam150a      mm1   \n",
       "56        intergenic_region  MODIFIER    Fam150a-St18      mm1   \n",
       "57           intron_variant  MODIFIER            St18      mm1   \n",
       "\n",
       "           CIRCOS_COLOR  RANDOM_NUMBER            COLOR  \\\n",
       "0   color=colorMODIFIER            184  color=colorCtoA   \n",
       "2   color=colorMODIFIER            342  color=colorCtoA   \n",
       "3   color=colorMODIFIER            965  color=colorAtoG   \n",
       "4   color=colorMODIFIER            508  color=colorTtoC   \n",
       "5   color=colorMODIFIER            370  color=colorTtoA   \n",
       "6   color=colorMODIFIER            456  color=colorAtoT   \n",
       "9   color=colorMODIFIER            215  color=colorCtoT   \n",
       "10  color=colorMODIFIER             17  color=colorCtoT   \n",
       "11  color=colorMODIFIER            124  color=colorGtoA   \n",
       "12  color=colorMODIFIER            179  color=colorTtoA   \n",
       "13  color=colorMODIFIER            354  color=colorAtoT   \n",
       "14  color=colorMODIFIER            549  color=colorCtoT   \n",
       "15  color=colorMODIFIER            667  color=colorCtoT   \n",
       "16  color=colorMODIFIER             30  color=colorGtoA   \n",
       "17  color=colorMODIFIER            116  color=colorTtoC   \n",
       "18  color=colorMODIFIER            441  color=colorTtoC   \n",
       "19  color=colorMODIFIER            767  color=colorTtoC   \n",
       "21  color=colorMODIFIER            486  color=colorAtoG   \n",
       "22  color=colorMODIFIER            554  color=colorGtoA   \n",
       "23  color=colorMODIFIER            236  color=colorTtoA   \n",
       "24  color=colorMODIFIER              3  color=colorGtoA   \n",
       "28  color=colorMODIFIER            547  color=colorCtoT   \n",
       "29  color=colorMODIFIER            946  color=colorAtoG   \n",
       "30  color=colorMODIFIER            511  color=colorAtoC   \n",
       "31  color=colorMODIFIER            513  color=colorGtoC   \n",
       "33  color=colorMODIFIER             14  color=colorAtoG   \n",
       "34  color=colorMODIFIER            852  color=colorTtoG   \n",
       "35  color=colorMODIFIER            448  color=colorAtoG   \n",
       "36  color=colorMODIFIER            272  color=colorCtoT   \n",
       "37  color=colorMODIFIER            733  color=colorGtoC   \n",
       "38  color=colorMODIFIER            660  color=colorTtoA   \n",
       "39  color=colorMODIFIER            335  color=colorTtoC   \n",
       "40  color=colorMODIFIER            725  color=colorGtoA   \n",
       "41  color=colorMODIFIER            178  color=colorAtoT   \n",
       "42  color=colorMODIFIER            362  color=colorTtoC   \n",
       "43  color=colorMODIFIER            228  color=colorTtoC   \n",
       "44  color=colorMODIFIER              5  color=colorCtoT   \n",
       "45  color=colorMODIFIER            932  color=colorCtoT   \n",
       "46  color=colorMODIFIER            909  color=colorGtoC   \n",
       "47  color=colorMODIFIER            526  color=colorTtoC   \n",
       "48  color=colorMODIFIER             56  color=colorCtoA   \n",
       "49  color=colorMODIFIER            997  color=colorAtoG   \n",
       "50  color=colorMODIFIER             34  color=colorCtoG   \n",
       "51  color=colorMODIFIER            502  color=colorTtoC   \n",
       "52  color=colorMODIFIER            995  color=colorTtoC   \n",
       "53  color=colorMODIFIER            336  color=colorAtoC   \n",
       "54  color=colorMODIFIER            969  color=colorCtoG   \n",
       "55  color=colorMODIFIER             98  color=colorGtoA   \n",
       "56  color=colorMODIFIER            495  color=colorCtoT   \n",
       "57  color=colorMODIFIER            873  color=colorCtoT   \n",
       "\n",
       "    Intermutation_Distance  Intermutation_Distance_log2  \n",
       "0                      NaN                          NaN  \n",
       "2                 228541.0                    17.802093  \n",
       "3                     32.0                     5.000000  \n",
       "4                   1415.0                    10.466586  \n",
       "5                  87045.0                    16.409474  \n",
       "6                   7773.0                    12.924256  \n",
       "9                 207721.0                    17.664288  \n",
       "10                     2.0                     1.000000  \n",
       "11                    76.0                     6.247928  \n",
       "12                    24.0                     4.584963  \n",
       "13                158944.0                    17.278159  \n",
       "14                     6.0                     2.584963  \n",
       "15                     4.0                     2.000000  \n",
       "16                     3.0                     1.584963  \n",
       "17                     4.0                     2.000000  \n",
       "18                    31.0                     4.954196  \n",
       "19                 57831.0                    15.819555  \n",
       "21                230383.0                    17.813675  \n",
       "22                     8.0                     3.000000  \n",
       "23                   265.0                     8.049849  \n",
       "24                    33.0                     5.044394  \n",
       "28                  2136.0                    11.060696  \n",
       "29                119585.0                    16.867677  \n",
       "30                291736.0                    18.154304  \n",
       "31                   674.0                     9.396605  \n",
       "33                    14.0                     3.807355  \n",
       "34                    34.0                     5.087463  \n",
       "35                    28.0                     4.807355  \n",
       "36                     8.0                     3.000000  \n",
       "37                 14223.0                    13.795938  \n",
       "38                459051.0                    18.808295  \n",
       "39                104306.0                    16.670463  \n",
       "40                 15544.0                    13.924070  \n",
       "41                 54801.0                    15.741915  \n",
       "42                119261.0                    16.863763  \n",
       "43                 94918.0                    16.534394  \n",
       "44                 66207.0                    16.014696  \n",
       "45                     9.0                     3.169925  \n",
       "46                     6.0                     2.584963  \n",
       "47                     2.0                     1.000000  \n",
       "48                    32.0                     5.000000  \n",
       "49                 30141.0                    14.879440  \n",
       "50                367366.0                    18.486859  \n",
       "51                 25124.0                    14.616779  \n",
       "52                     4.0                     2.000000  \n",
       "53                     5.0                     2.321928  \n",
       "54                191323.0                    17.545651  \n",
       "55                438244.0                    18.741375  \n",
       "56                 49043.0                    15.581760  \n",
       "57                 89005.0                    16.441599  "
      ]
     },
     "execution_count": 27,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "df = df.sort_values(by=[\"chrom\",\"position\"])\n",
    "print(df[\"position\"].dtype)\n",
    "df[\"position\"] = df[\"position\"].astype(float)\n",
    "df['Intermutation_Distance'] = abs(df['position'] - df['position'].shift(1))\n",
    "df['Intermutation_Distance_log2'] = np.log2(df['Intermutation_Distance'])\n",
    "df.head(50)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "d74f5e3e",
   "metadata": {},
   "outputs": [],
   "source": []
  }
 ],
 "metadata": {
  "kernelspec": {
   "display_name": "Python 3 (ipykernel)",
   "language": "python",
   "name": "python3"
  },
  "language_info": {
   "codemirror_mode": {
    "name": "ipython",
    "version": 3
   },
   "file_extension": ".py",
   "mimetype": "text/x-python",
   "name": "python",
   "nbconvert_exporter": "python",
   "pygments_lexer": "ipython3",
   "version": "3.9.7"
  }
 },
 "nbformat": 4,
 "nbformat_minor": 5
}
